tests/v2/test_waku_pagination.nim

```diff
@@ -15,14 +25,193 @@ procSuite "pagination":
       len(index.digest.data) == 32 # sha2 output length in bytes
       index.receivedTime != 0 # the timestamp should be a non-zero value
 
-  test "computeIndex: identical WakuMessages test":
     let
-      wm = WakuMessage(payload: @[byte 1, 2, 3], contentTopic: ContentTopic(1))
-      index1 = wm.computeIndex()
+      wm1 = WakuMessage(payload: @[byte 1, 2, 3], contentTopic: ContentTopic(1))
+      index1 = wm1.computeIndex()
       wm2 = WakuMessage(payload: @[byte 1, 2, 3], contentTopic: ContentTopic(1))
       index2 = wm2.computeIndex()
 
     check:
       # the digests of two identical WakuMessages must be the same
       index1.digest == index2.digest
 
+  test "Index comparison, IndexedWakuMessage comparison, and Sorting tests":
+    var data1 {.noinit.}: array[32, byte]
+    for x in data1.mitems: x = 1
+    var data2 {.noinit.}: array[32, byte]
+    for x in data2.mitems: x = 2
+    var data3 {.noinit.}: array[32, byte]
+    for x in data3.mitems: x = 3
+      
+    let
+      index1: Index = Index(receivedTime: 1, digest: MDigest[256](data: data1))
+      index2: Index = Index(receivedTime: 1, digest: MDigest[256](data: data2))
+      index3: Index = Index(receivedTime: 2, digest: MDigest[256](data: data3))
```

Comment:
```suggestion
      index1 = Index(receivedTime: 1, digest: MDigest[256](data: data1))
      index2 = Index(receivedTime: 1, digest: MDigest[256](data: data2))
      index3 = Index(receivedTime: 2, digest: MDigest[256](data: data3))
```

Reply:
While the type inference absolutely works here and indeed is nicer, but curious to know whether there might be any compilation/performance issue when the explicit declaration is used?